### PR TITLE
[IMP] point_of_sale: show combo line amount and hide individual line amounts

### PR DESCRIPTION
**Pull request by parp-odoo**
In this commit:
===============
- Display the total amount for combo product lines based on selected product prices.
- Hide the amounts for individual order lines that are part of a combo.

task- 4229373

### CODE REVIEW

**Review by adgu-odoo — CHANGES_REQUESTED** [review]
After discussion with MOBT, could you also remove the price on the combo lines on the customer screen and the order widget. So I guess the best solution would maybe to change the `getPriceString` method ?

**Comment by parp-odoo** [comment]
@adgu-odoo ,
Changes are done!
Could you please check?

**Comment by caburj** [comment]
rebase please

**Comment by parp-odoo** [comment]
@caburj ,
Done

File: addons/point_of_sale/static/src/app/models/pos_order_line.js [hunk 1]
```diff
@@ -366,11 +366,23 @@ export class PosOrderline extends Base {
...
-        }
+        const prices =
+            this.combo_line_ids.length > 0
+                ? this.combo_line_ids.reduce(
+                      (acc, cl) => {
+                          const clPrices = cl.getAllPrices(1);
+                          return {
+                              priceWithTax: acc.priceWithTax + clPrices.priceWithTax,
+                              priceWithoutTax: acc.priceWithoutTax + clPrices.priceWithoutTax,
+                          };
```
Comment (adgu-odoo) [inline addons/point_of_sale/static/src/app/models/pos_order_line.js:377]:
```suggestion
                          return {
                              priceWithTax: acc.priceWithTax + clPrices.allUnitPrices.priceWithTax,
                              priceWithoutTax: acc.priceWithoutTax + clPrices.allUnitPrices.priceWithoutTax,
                          };
```

File: addons/point_of_sale/static/src/app/models/pos_order_line.js [hunk 2]
```diff
@@ -366,11 +366,23 @@ export class PosOrderline extends Base {
...
+                      },
+                      { priceWithTax: 0, priceWithoutTax: 0 }
+                  )
+                : this.getAllPrices(1);
```
Comment (adgu-odoo) [inline addons/point_of_sale/static/src/app/models/pos_order_line.js:381]:
```suggestion
                : this.allUnitPrices;
```

Comment (adgu-odoo) [reply addons/point_of_sale/static/src/app/models/pos_order_line.js:381]: `allUnitPrices` is cached when called, so calling it is better than calling `getAllPrices(1)` as if the value did not changed, we should not recompute it

**Review by adgu-odoo — APPROVED** [review]

**Comment by adgu-odoo** [comment]
robodoo r+

**Pull request closed without merging**